lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -557,6 +558,25 @@ def confirm_namespace_exists
       @logger.info("Namespace #{@namespace} found")
     end
 
+    # make sure to never prune the ejson-keys secret
+    def confirm_ejson_keys_not_prunable
+      raw_secret, st, err = nil
+      with_retries(2) do
+        raw_secret, err, st = kubectl.run("get", "secret",
+          EjsonSecretProvisioner::EJSON_KEYS_SECRET, "-oyaml", output_is_sensitive: true)
+        st.success? || err.include?(KubernetesDeploy::Kubectl::NOT_FOUND_ERROR_TEXT)
+      end
+      if st.success?
+        secret = YAML.safe_load(raw_secret, [Date, Time])
+        return unless secret.dig("metadata", "annotations", LAST_APPLIED_ANNOTATION)
+
+        raise EjsonPrunableError,
+          "Found #{LAST_APPLIED_ANNOTATION} annotation on #{EjsonSecretProvisioner::EJSON_KEYS_SECRET} secret. " \
+          "kubernetes-deploy will not continue since it is extremely unlikely that this secret should be pruned. " \
+          "contact your cluster administrator for further assistance."
```

Comment:
Nit: capitalize this sentence

Reply:
Passing as an error message downcases everything except the first char. Moving this to a logger entry and then raising an empty error

Reply:
Ok, but make sure it is still the summary section that is surfacing it

Reply:
I added an error log to phase 1 when the secret is prunable and moved the original error back into the exception object. I simply removed the `contact your cluster administrator for further assistance` since it's pretty redundant.